src/Hosting/Abstractions/src/Microsoft.AspNetCore.Hosting.Abstractions.csproj

```diff
@@ -11,6 +11,28 @@
     <IsTrimmable>true</IsTrimmable>
   </PropertyGroup>
 
+  <ItemGroup>
+    <Compile Include="$(SharedSourceRoot)Metrics\**\*.cs" LinkBase="Metrics" />
+  </ItemGroup>
+
+  <!-- Temporary hack to make prototype Metrics DI integration types available -->
```

Comment:
Is there a bug tracking this? This is a pretty massive hack.

Reply:
https://github.com/dotnet/aspnetcore/issues/47618